lib/main_common.pm

```diff
@@ -2400,117 +2400,109 @@ sub load_virtualization_tests {
+                    loadtest "virtualization/universal/list_guests";
+                    loadtest "virtualization/universal/patch_guests";
+                } elsif ($update_package eq 'xen' || $update_package eq 'qemu') {
+                    loadtest "virt_autotest/login_console";
+                    loadtest "virtualization/universal/list_guests";
+                }
+            }
+        }
+        loadtest "virtualization/universal/kernel";
+        loadtest "virtualization/universal/finish";
+    }
+
+    my @virt_functions = (
+        ['ENABLE_SAVE_AND_RESTORE', 'virtualization/universal/save_and_restore'],
+        ['ENABLE_GUEST_MANAGEMENT', 'virtualization/universal/guest_management'],
+        ['ENABLE_FINAL', 'virtualization/universal/ssh_final,virtualization/universal/virtmanager_final,virtualization/universal/smoketest,virtualization/universal/stresstest,console/perf'],
+        ['ENABLE_STORAGE', 'virtualization/universal/storage'],
+        ['ENABLE_WINDOWS', 'virtualization/universal/download_image,virtualization/universal/windows'],
+        ['ENABLE_DOM_METRICS', 'virtualization/universal/virsh_stop,virtualization/universal/xl_create,virtualization/universal/dom_install,virtualization/universal/dom_metrics,virtualization/universal/xl_stop,virtualization/universal/virsh_start', 'xen'],
+        ['ENABLE_IRQBALANCE', 'virt_autotest/xen_guest_irqbalance', 'xen'],
+        ['ENABLE_VIR_NET', 'virt_autotest/libvirt_host_bridge_virtual_network,virt_autotest/libvirt_nated_virtual_network,virt_autotest/libvirt_isolated_virtual_network'],
+        ['ENABLE_VIRTMANAGER', 'virtualization/universal/virtmanager_init,virtualization/universal/virtmanager_offon,virtualization/universal/virtmanager_add_devices,virtualization/universal/virtmanager_rm_devices'],
+        ['ENABLE_HOTPLUGGING', 'virtualization/universal/hotplugging_guest_preparation,virtualization/universal/hotplugging_network_interfaces,virtualization/universal/hotplugging_HDD,virtualization/universal/hotplugging_vCPUs,virtualization/universal/hotplugging_memory,virtualization/universal/hotplugging_cleanup'],
+        ['ENABLE_SNAPSHOTS', 'virt_autotest/virsh_internal_snapshot,virt_autotest/virsh_external_snapshot', 'kvm'],
+        ['ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH', 'virt_autotest/sriov_network_card_pci_passthrough'],
+    );
 
-    if (check_var('VIRT_PART', 'final')) {
-        loadtest 'virtualization/universal/ssh_final';    # Check that every guest is reachable over SSH
-        loadtest 'virtualization/universal/virtmanager_final';    # Check that every guest shows the login screen
-        loadtest "virtualization/universal/smoketest";    # Virtualization smoke test for hypervisor
-        loadtest "virtualization/universal/stresstest";    # Perform stress tests on the guests
-        loadtest "console/perf";
-        loadtest "console/oprofile" unless (get_var("REGRESSION", '') =~ /xen/);
+    if (get_var('UPDATE_PACKAGE') eq 'kernel-default') {
```

Comment:
Hi Roy,
Is this part necessary? In A2 task of sriov, in its description, it mentions that a new testsuite named qam-kvm-install-and-sriov-test will be introduced. From A5 task, you can see  below. So kernel package will run this qam-kvm-install-and-sriov-test testsuite. Then setting all these feature switch value to 0 can actually be done just within the testsuite settings. 

```
 elsif UPDATE_PACKAGE == kernel-default
        run kvm/xen: sriov test(includes host&guest installation) 
            vmware test,
            hyperv test
```

Reply:
this is my testsuit:
qam-kvm-install-and-features-test
`AUTOYAST=1
DESKTOP=gnome
ENABLE_DOM_METRICS=1
ENABLE_FINAL=1
ENABLE_GUEST_MANAGEMENT=1
ENABLE_HOST_INSTALLATION=1
ENABLE_HOTPLUGGING=1
ENABLE_IRQBALANCE=1
ENABLE_SAVE_AND_RESTORE=1
ENABLE_SNAPSHOTS=1
ENABLE_STORAGE=1
ENABLE_VIRTMANAGER=1
ENABLE_VIR_NET=1
ENABLE_VM_INSTALL=1
ENABLE_WINDOWS=1
HOST_HYPERVISOR=kvm
IPMI_BACKEND_MC_RESET=1
IPMI_DO_NOT_RESTART_HOST=1
KVM=1
LTSS_REGCODES_SECRET=xxxx
MAX_JOB_TIME=43200
REGRESSION=kvm
SYSTEM_ROLE=kvm
VIRT_IMAGE_PATH=virt-nfs.qam.suse.cz:/virt_nfs/qam-virt
_SECRET_RSA_PRIV_KEY=-----BEGIN OPENSSH PRIVATE KEY----
_SECRET_RSA_PUB_KEY=ssh-rsa `

qam-kvm-install-and-sriov-test
`AUTOYAST=1
DESKTOP=gnome
ENABLE_HOST_INSTALLATION=1
ENABLE_SRIOV=1
ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH=1
ENABLE_VM_INSTALL=1
HOST_HYPERVISOR=kvm
IPMI_BACKEND_MC_RESET=1
IPMI_DO_NOT_RESTART_HOST=1
KVM=1
LTSS_REGCODES_SECRET=XXX
MAX_JOB_TIME=43200
REGRESSION=kvm
SYSTEM_ROLE=kvm
VIRT_IMAGE_PATH=virt-nfs.qam.suse.cz:/virt_nfs/qam-virt
_SECRET_RSA_PRIV_KEY=-----BEGIN OPENSSH PRIVATE KEY---
_SECRET_RSA_PUB_KEY=ssh-rsa `

Reply:
Let me think about it carefully.

Reply:
I think the default given values in testsuite is good to be what we want by default. That is , for qam-kvm-install-and-features-test

```
[qam-kvm-install-and-features-test]

ENABLE_SRIOV=1 => change to 0
ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH=1  => change to 0, better to have only one switch for SRIOV, so merge the two into one?






[qam-kvm-install-and-sriov-test]

...
# add sriov switch
ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH=1 

# update below feature switches to 0
ENABLE_DOM_METRICS=1
ENABLE_FINAL=1
ENABLE_GUEST_MANAGEMENT=1
ENABLE_HOTPLUGGING=1
ENABLE_IRQBALANCE=1
ENABLE_SAVE_AND_RESTORE=1
ENABLE_SNAPSHOTS=1
ENABLE_STORAGE=1
ENABLE_VIRTMANAGER=1
ENABLE_VIR_NET=1
ENABLE_WINDOWS=1
```

Reply:
When update_package is kernel default, trigger testsuit: qam-kvm(xen)-install-and-sriov-test
When update_package is other option, trigger testsuit: qam-kvm(xen)-install-and-sriov-test(sometimes no sriov test, just only package installation test) and qam-kvm(xen)-install-and-features-test.

Reply:
VR testing after new parameter configuration.

Reply:
> When update_package is kernel default, trigger testsuit: qam-kvm(xen)-install-and-sriov-test When update_package is other option, trigger testsuit: qam-kvm(xen)-install-and-sriov-test(sometimes no sriov test, just only package installation test) and qam-kvm(xen)-install-and-features-test.

@RoyCai7  Just to sync understanding. It seems to me that with below  test need (described in A5 ticket), we won't need to run qam-kvm(xen)-install-and-sriov-test( no sriov test). Right? Or do I overlook anything?

```
    if UPDATE_PACKAGE == qemu/libvirt-daemon/
        run kvm/xen: installation and feature tests, sriov test, migration test
    elsif UPDATE_PACKAGE == xen
        run xen: installation and feature tests, sriov test, migration test
    elsif UPDATE_PACKAGE == virt-manager,libguestfs,libslirp
        run kvm/xen: installation and feature tests
    elsif UPDATE_PACKAGE == open-vm-tools
        run vmware test only
    elsif UPDATE_PACKAGE == kernel-default
        run kvm/xen: sriov test(includes host&guest installation) 
            vmware test,
            hyperv test
```

Reply:
yes,  2(TD or not, only 12sp3/15sp4 =2, others 1) * 5 flavors * 7 products (12sp3,12sp5,15sp2~sp6,...)
when start a specific flavor, it will schedule multi test suite.

Reply:
VR testing.

Reply:
Test completed, please check again.
[Sriov test](http://10.100.103.128/tests/575)
[Feature test](http://10.100.103.128/tests/574) there are 2 bugs: [1221917](https://bugzilla.suse.com/show_bug.cgi?id=1221917) and [1201778](https://bugzilla.suse.com/show_bug.cgi?id=1201778)
I filed 2 tickets to set it softfailed: [162053](https://progress.opensuse.org/issues/162056) and [162053](https://progress.opensuse.org/issues/162053)

Reply:
Thank you @RoyCai7 . I see a weird thing from the  verification links. It seems the switch settings are not acting as expected. In that sriov test link, the ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH is 0. For feature test verification link -- http://10.100.103.128/tests/574#settings, the switches  for all features are 0 except that ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH is 1(it shall not), but from test modules, most of the feature tests are actually run... Why?
One more reminder for verification check detail: please check if snapshot tests are loaded as the *last* in testsuite `qam-xen-install-and-features-test`.

Reply:
@RoyCai7  Did you miss above comment?

Reply:
XEN has no snapshot test, only kvm has such snapshot test.
XEN history test: http://openqa.qam.suse.cz/tests/75013#dependencies
KVM history test: http://openqa.qam.suse.cz/tests/74809#dependencies

Reply:
still checking, latest reply to it.

Reply:
0 stands for test.
1or no such parameter stands for not test.

For sriov test: 
ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH	0
only test this module


For feature test, not test sriov
ENABLE_DOM_METRICS	0
ENABLE_FINAL	0
ENABLE_GUEST_MANAGEMENT	0
ENABLE_HOTPLUGGING	0
ENABLE_IRQBALANCE	0
ENABLE_SAVE_AND_RESTORE	0
ENABLE_SNAPSHOTS	0
ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH	1
ENABLE_STORAGE	0
ENABLE_VIRTMANAGER	0
ENABLE_VIR_NET	0
ENABLE_WINDOWS	0
will test modules except ENABLE_SRIOV_NETWORK_CARD_PCI_PASSTHROUGH

I think it is reasonable and expected.

Reply:
I see, so you are using '0' to enable feature test, while '1' to disable it. Root cause is this line.
```
  if (check_var($module_switch, 0) && (!$host_hypervisor || check_var('HOST_HYPERVISOR', $host_hypervisor))) {
```

This is the opposite meaning of the var name. It is  also the opposite usage with other switches like `ENABLE_HOST_INSTALLATION` and `ENABLE_VM_INSTALL`, for which 1 means to enable test, and 0 means to disable test. Why not make it aligned on usage of feature switches -- 1 means to enable, 0 means to disable?

Reply:
1 means to enable, 0 or no such parameter means to disable.
updated code and test suites
VR testing.

Reply:
[Sriov test](http://10.100.103.128/tests/577)
[Feature test](http://10.100.103.128/tests/576) there are 2 bugs: [1221917](https://bugzilla.suse.com/show_bug.cgi?id=1221917) and [1201778](https://bugzilla.suse.com/show_bug.cgi?id=1201778)
I filed 2 tickets to set it softfailed: [162053](https://progress.opensuse.org/issues/162056) and [162053](https://progress.opensuse.org/issues/162053)

Reply:
> [Sriov test](http://10.100.103.128/tests/578)

This one looks good.

> [Feature test](http://10.100.103.128/tests/576) 

This seems problematic: snapshot is missing although the switch for it is set to 1.

>there are 2 bugs: [1221917](https://bugzilla.suse.com/show_bug.cgi?id=1221917) and [1201778](https://bugzilla.suse.com/show_bug.cgi?id=1201778) I filed 2 tickets to set it softfailed: [162053](https://progress.opensuse.org/issues/162056) and [162053](https://progress.opensuse.org/issues/162053)

Thanks for the 2 tickets. I bumped their priority to `Urgent`. Also we need to hold this PR's merge until those 2 tickets are merged. Otherwise we can't do incident test with the new testsuite introduced in this PR.

Reply:
This is xen test, no snapshot test.  it is expected.
['ENABLE_SNAPSHOTS', 'virt_autotest/virsh_internal_snapshot,virt_autotest/virsh_external_snapshot', 'kvm'],

More test for kvm:
http://10.100.103.128/tests/579

Reply:
> This is xen test, no snapshot test. it is expected. ['ENABLE_SNAPSHOTS', 'virt_autotest/virsh_internal_snapshot,virt_autotest/virsh_external_snapshot', 'kvm'],
> 
> More test for kvm: [KVM Feature test](http://10.100.103.128/tests/579)

@RoyCai7 Hi Roy, xen supports sriov. See initial automation ticket https://progress.opensuse.org/issues/89098 and this OSD 15sp6 xen sriov test https://openqa.suse.de/tests/14610205.
Are we missing sriov for xen incidents?  It seems so from Ongoing Runs page. If yes, we need to correct that.